gossip-base/src/main/java/org/apache/gossip/transport/AbstractTransportManager.java

```diff
@@ -77,11 +75,9 @@ public void shutdown() {
 
   @Override
   public void startActiveGossiper() {
-    activeGossipThread.init(); 
+    activeGossipThread.init();
   }
 
   @Override
-  public void startEndpoint() {
-    gossipThreadExecutor.execute(passiveGossipThread);
-  }
+  public abstract void startEndpoint();
```

Comment:
It is a nice hook to start the thread outside of the constructor so it seems useful. Generally we have tried to avoid anything in the constructor that could throw and favored init()